examtool/examtool/api/convert.py

```diff
@@ -503,17 +488,29 @@ def import_file(filepath: str) -> str:
         return f.read()
 
 
-def handle_imports(buff: LineBuffer, path: str):
-    while not buff.empty():
-        line = buff.pop()
-        mode, directive, rest = parse_directive(line)
-        if mode == "IMPORT":
-            buff.remove_prev()
-            filepath = " ".join([directive, rest]).rstrip()
-            if path:
-                filepath = os.path.join(path, filepath)
-            new_buff = LineBuffer(import_file(filepath))
-            folderpath = os.path.dirname(filepath)
-            handle_imports(new_buff, folderpath)
-            buff.insert_next(new_buff)
-    buff.reset()
+def load_imports(base_src: str, base_path: str):
+    lines = []
+
+    def _load(src: str, path: str):
+        for i, line in enumerate(src.split("\n")):
+            mode, directive, rest = parse_directive(line)
+            if mode == "IMPORT":
+                filepath = os.path.join(
+                    dirname(path), " ".join([directive, rest]).rstrip()
+                )
+                try:
+                    _load(import_file(filepath), filepath)
+                except FileNotFoundError:
+                    raise SyntaxError(f"Unable to import {filepath}")
```

Comment:
This is not a good error message since it does not tell which file/line it is on. Can you add the file/line it was found on? Probably something like:
```python3
f"Parse stopped on {parentfilepath}:{i + 1} with error Unable to import {filepath}"
```